### PR TITLE
Add pyflakes check in Github actions

### PR DESCRIPTION
**Pull request by RemiLehe**

### CODE REVIEW

**Comment by RemiLehe** [comment]
Needs #4233 to fix `pyflakes` errors.

**Comment by ax3l** [comment]
@RemiLehe can we add this to `.pre-commit-config.yaml` instead, ideally if it supports auto-fixes?

**Comment by ax3l** [comment]
Ah, we found no pre-commit for it.
Since pyflake only lints Python code but has no auto-fixes, we will add it as currently proposed :)

**Review by ax3l — APPROVED** [review]

**Comment by ax3l** [comment]
Ah no, I found a way to keep it in pre-commit (w/o auto fixes): https://github.com/AMReX-Codes/pyamrex/pull/208

File: .github/workflows/source.yml [hunk 1]
```diff
@@ -13,8 +13,16 @@ concurrency:
...
 jobs:
-  style:
+  lint:
+    runs-on: ubuntu-22.04
+
+    steps:
+    - name: check WarpX python code
+      run: |
+        python3 -m pip install pyflakes
+        pyflakes .
```
Comment (ax3l) [inline .github/workflows/source.yml]:
```suggestion
        python3 -m pyflakes .
```

File: .github/workflows/source.yml [hunk 2]
```diff
@@ -13,8 +13,16 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  style:
+  lint:
+    runs-on: ubuntu-22.04
+
+    steps:
+    - name: check WarpX python code
```
Comment (ax3l) [inline .github/workflows/source.yml:21]: I forgot to get the source code:
```suggestion
    - uses: actions/checkout@v3
    - name: check WarpX python code
```

**Comment by ax3l** [comment]
Let use use `ruff` as in ImpactX & pyAMReX - it has the same job but is very fast :)
https://github.com/ECP-WarpX/impactx/blob/4676a26d4d14a7cc099ad63abbd703f9b2297726/.pre-commit-config.yaml#L95-L102

**Pull request closed without merging**